### PR TITLE
Remove users existing subscriptions on transition

### PR DESCRIPTION
**Pull request by gsvr**
- [x] Opt-out
- [x] SMS line

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -715,6 +715,14 @@ go.utils = {
     },
 
     control_api_call: function (method, payload, endpoint, im) {
+    // george: alt1 & alt2 were born from the PUT request.  I wasn't sure if we needed to pass in the
+    // params again for the query.  Couldn't find good resource on PUT usage.  Current code assumes we
```
Comment (imsickofmaps) [inline src/utils.js]: No querystring params required for PUT request

File: src/utils.js [hunk 2]
```diff
@@ -725,10 +733,24 @@ go.utils = {
           case "post":
             return http.post(im.config.control.url + endpoint, {
                 data: JSON.stringify(payload)
+                // alt2
+                // data: JSON.stringify(opts.payload)
               });
           case "get":
             return http.get(im.config.control.url + endpoint, {
-                params: payload
+                params: payload // george: JSON.stringify? don't think so.
```
Comment (imsickofmaps) [inline src/utils.js]: No - the helper handles that

File: src/optout.js [hunk 3]
```diff
@@ -109,7 +106,21 @@ go.app = function() {
...
+                    'state:enter': function() {
+                        // george: should we also run subscription_unsubscribe_all here?
+                        // george: and change the subscription type and frequency to 'none' / '0'?
+                        // george: I think this depends on how we want to deal with people opting back in -
+                        //         do they have to register again or continue where they left off?
+                        return self.im.api_request('optout.optout', {
+                            address_type: "msisdn",
+                            address_value: self.im.user.addr,
+                            message_id: self.im.msg.message_id
+                        });
```
Comment (imsickofmaps) [inline src/optout.js]: Chain the `then` here

File: go-app-smsinbound.js [hunk 4]
```diff
@@ -1038,13 +1079,19 @@ go.app = function() {
...
-                        return Q.all([
-                            go.utils.subscription_send_doc(self.contact, self.im, self.metric_prefix, opts),
-                            self.im.contacts.save(self.contact)
-                        ]);
+
+                        return go.utils
+                            .subscription_unsubscribe_all(self.contact, self.im, opts)
+                            .then(function() {
+                                return Q.all([
+                                    // george: should we be notifying jembi of birth?
```
Comment (imsickofmaps) [inline go-app-smsinbound.js]: Not now. Email sent to Debbie to ask about future.

**Comment by gsvr** [comment]
@imsickofmaps Please review.

**Comment by imsickofmaps** [comment]
:+1:

**Pull request merged**